ma/maInput.h

```diff
@@ -165,21 +168,25 @@ Input* configure(
  \param size a scalar field of desired element size
  \param s if non-zero, use that to transfer all fields. otherwise,
           transfer any associated fields with default algorithms */
-Input* configure(
+const Input* configure(
     Mesh* m,
     apf::Field* size,
     SolutionTransfer* s=0);
 
 /** \brief generate a uniform refinement configuration */
-Input* configureUniformRefine(Mesh* m, int n=1, SolutionTransfer* s=0);
+const Input* configureUniformRefine(Mesh* m, int n=1, SolutionTransfer* s=0);
 /** \brief generate a matched uniform refinement configuration */
-Input* configureMatching(Mesh* m, int n=1, SolutionTransfer* s=0);
+const Input* configureMatching(Mesh* m, int n=1, SolutionTransfer* s=0);
 /** \brief generate a no-op configuration */
-Input* configureIdentity(Mesh* m, SizeField* f=0, SolutionTransfer* s=0);
+const Input* configureIdentity(Mesh* m, SizeField* f=0, SolutionTransfer* s=0);
 
 /** \brief used internally, but users can call this if they want */
 void validateInput(Input* in);
 
+/** \brief removes constant-ness from a constant Input pointer
+           so users can modify it */
+Input* makeAdvanced(const Input* in);
```

Comment:
Since we are stuck using pointers for Input the semantics is unclear. This requires a comment that original the input gets put into a invalid state after this function is called.

Reply:
updated this.